_docker/drupal/drupal-filesystem/web/modules/custom/rhd_katacoda/rhd_katacoda.tokens.inc

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * @file
+ * Implements custom token functionality
+ * 
+ * This creates a token specifically to show a "parent" content type for a "child"
+ * content type's URL alias. The content type name and field/machine names must be copied 
+ * from the RHD Site in order for this to function properly.
+ * 
+ * To create this custom token, 4 things must happen:
+ *
+ * 1. Two content-types must exist, with one being the "parent" of the other
+ *    by setting an entity reference in the parent field to find entries for the content type of the "child"
+ * 2. The "parent" content type must have a field for a (custom URL slug)
+ * 3. The slug name must exist in order for the token placement to work successfully.
+ * 4. Once the alias is created, the token 'rhd_katacoda_parent_course_token' (or whatever is preferred) must be used as a the alias value.
+ *  
+ */
+
+use Drupal\Core\Render\BubbleableMetadata;
+
+/**
+ * Implements hook_token_info()
+ * 
+ * Declare a token name as part of the node.
+ * No new token group is declared since some other node values (namely, the :nid) are needed when the nodes
+ * are created
+ */
+
+ function rhd_katacoda_token_info() {
+    $info['tokens']['node']['rhd_katacoda_parent_course_token'] = [
+        'name'=>t('RHD Katacoda Parent Course'),
+        'description'=>t('This is the URL slug for the parent course of a child lesson entity reference')];
+      return $info;
+ }
+ 
+ /**
+  * Implements hook_tokens().
+  * 
+  * Find the parent content type course entry of a child content type lesson entry by
+  * way of a data query and insert the URL slug value of the parent as the rendered.
```

Comment:
Very unimportant, but is the period at the end of this line intended?

Reply:
Huh... no. Not important at all. Not sure why I'd have added that.

Reply:
Maybe I was just in "writing mode" vs. "code mode" when adding these comments. Period would be force of habit.